source/renderer/app/containers/staking/dialogs/UndelegateDialogContainer.js

```diff
@@ -0,0 +1,97 @@
+// @flow
+import React, { Component } from 'react';
+import { inject, observer } from 'mobx-react';
+import type { StoresMap } from '../../../stores/index';
+import type { ActionsMap } from '../../../actions/index';
+import UndelegateConfirmationDialog from '../../../components/staking/delegation-center/UndelegateConfirmationDialog';
+import UndelegateConfirmationResultDialog from '../../../components/staking/delegation-center/UndelegateConfirmationResultDialog';
+
+type Props = {
+  stores: any | StoresMap,
+  actions: any | ActionsMap,
+  onExternalLinkClick: Function,
+};
+
+@inject('actions', 'stores')
+@observer
+export default class UndelegateDialogContainer extends Component<Props> {
+  static defaultProps = { actions: null, stores: null };
+
+  render() {
+    const { actions, stores, onExternalLinkClick } = this.props;
+    const { uiDialogs, wallets } = stores;
+    const dialogData = uiDialogs.dataForActiveDialog;
+    const { walletId } = dialogData;
+    const {
+      quitStakePoolRequest,
+      getWalletById,
+      undelegateWalletSubmissionSuccess,
+    } = wallets;
+    const walletToBeUndelegated = getWalletById(walletId);
+
+    if (!walletToBeUndelegated) {
+      throw new Error(
+        'Delegated wallet should be selected for UndelegateDialogContainer.'
```

Comment:
@yakovkaravelov not sure if this error is just added for protection or if we ever expect to show it. In case this error could be displayed we need to improve it and also add Japanese translation for it.

Reply:
@nikolaglumac This is going to be taken care of once @tomothespian and @daniloprates 's PRs are finished. They are going to be used in this PR for updated approach for stake pool and next epoch info, etc.

Reply:
OK